### PR TITLE
feat: use aggregation for lots by year created filter

### PR DESCRIPTION
**Pull request by egdbear**
The type of this PR is: **Feat**

This PR solves [DIA-106]

### Description

cc @artsy/diamond-devs 

This PR uses new `LOTS_BY_CREATED_YEAR` aggregation from diffusion (filters endpoint).

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DIA-106]: https://artsyproduct.atlassian.net/browse/DIA-106?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx [hunk 1]
```diff
@@ -383,8 +394,8 @@ export const ArtistAuctionResultsRefetchContainer = createRefetchContainer(
         aggregations={
           props.aggregations as SharedAuctionResultsFilterContextProps["aggregations"]
         }
-        earliestCreatedYear={startAt}
-        latestCreatedYear={endAt}
+        earliestCreatedYear={startAt ? parseInt(startAt) : null}
```
Comment (araujobarret) [inline src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx:397]: is it possible to return a NaN from the parse?

File: src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/YearCreated.tsx [hunk 2]
```diff
@@ -1,66 +1,45 @@
...
   const {
     createdAfterYear,
     createdBeforeYear,
     allowEmptyCreatedDates,
   } = useCurrentlySelectedFiltersForAuctionResults()
 
-  const hasChanges =
-    earliestCreatedYear !== createdAfterYear ||
-    latestCreatedYear !== createdBeforeYear
+  const options = (
```
Comment (araujobarret) [inline src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/YearCreated.tsx:19]: it looks way more sturdy 💪

**Review by araujobarret — DISMISSED** [review]
left a minor Q(non-blocking), it looks good 👍

File: src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx (hunk 1 above)
Comment (egdbear) [reply src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx:397]: Nope we are not aggregating null values [see here](https://github.com/artsy/diffusion/blob/main/app/queries/lot_filtered_query.rb#L51)

**Pull request merged**